### ISSUE TITLE
Stuck at boot

### ISSUE THREAD

**Issue by abyszuk** [ikwzm/ZynqMP-FPGA-Ubuntu22.04-Desktop#1]
Hello,

I've been playing with your image and have trouble running it on my KR260 devkit.
The boot always hangs, the last kernel messages are:
```
[    3.836592] macb ff0b0000.ethernet: Not enabling partial store and forward
[    3.838795] of_cfs_init: OK
[    3.845701] macb ff0b0000.ethernet: invalid hw address, using random
[    3.848508] cfg80211: Loading compiled-in X.509 certificates for regulatory database
[    3.855352] macb ff0b0000.ethernet eth0: Defer probe as mdio producer ff0c0000.ethernet is not probed
[    3.863854] cfg80211: Loaded X.509 cert 'sforshee: 00b28ddf47aef9cea7'
[    3.878297] clk: Not disabling unused clocks
[    3.882817] ALSA device list:
[    3.885775]   #0: DP mon
[    3.888982] platform regulatory.0: Direct firmware load for regulatory.db failed with error -2
[    3.897603] cfg80211: failed to load regulatory.db
[    3.927760] macb ff0c0000.ethernet: Not enabling partial store and forward
[    3.934673] macb ff0c0000.ethernet: invalid hw address, using random
[    3.999594] Waiting for root device /dev/sda2...
[    3.999771] macb ff0b0000.ethernet: Not enabling partial store and forward
[    4.011104] macb ff0b0000.ethernet: invalid hw address, using random
[    4.017955] macb ff0b0000.ethernet eth0: Defer probe as mdio producer ff0c0000.ethernet is not probed
[    4.043550] macb ff0c0000.ethernet: Not enabling partial store and forward
[    4.050447] macb ff0c0000.ethernet: invalid hw address, using random
[    4.409729] zynqmp-display fd4a0000.display: [drm] Cannot find any crtc or sizes
[   14.046087] macb ff0b0000.ethernet: Not enabling partial store and forward
[   14.052996] macb ff0b0000.ethernet: invalid hw address, using random
[   14.059897] macb ff0b0000.ethernet eth0: Defer probe as mdio producer ff0c0000.ethernet is not probed
[   14.083611] macb ff0c0000.ethernet: Not enabling partial store and forward
[   14.090510] macb ff0c0000.ethernet: invalid hw address, using random
[   14.127895] platform ff0b0000.ethernet: deferred probe pending
[   14.133739] platform ff0c0000.ethernet: deferred probe pending
[   14.139564] platform ff9d0000.usb: deferred probe pending
[   14.144954] platform ff9e0000.usb: deferred probe pending
[   14.150344] platform pwm-fan: deferred probe pending
```
I think the main problem is hinted by `waiting for root device /dev/sda2`.
For a reference, official Ubuntu image works fine.

**Comment by ikwzm** [comment]
Thank you for the issue.

As you guessed, the usb storage does not seem to be working. 
If it were working properly, you would get a message like this

```console
[    3.268511] Waiting for root device /dev/sda2...
[    3.277004] cfg80211: failed to load regulatory.db
[    3.363645] usb 1-1: New USB device found, idVendor=0424, idProduct=2744, bcdDevice= 2.21
[    3.371834] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    3.378968] usb 1-1: Product: USB2744
[    3.382625] usb 1-1: Manufacturer: Microchip Tech
[    3.387434] usb 3-1: new high-speed USB device number 2 using xhci-hcd
[    3.442614] hub 1-1:1.0: USB hub found
[    3.446395] hub 1-1:1.0: 4 ports detected
[    3.506492] usb 2-1: new SuperSpeed USB device number 2 using xhci-hcd
[    3.531442] usb 2-1: New USB device found, idVendor=0424, idProduct=5744, bcdDevice= 2.21
[    3.539625] usb 2-1: New USB device strings: Mfr=2, Product=3, SerialNumber=0
[    3.546759] usb 2-1: Product: USB5744
[    3.550422] usb 2-1: Manufacturer: Microchip Tech
[    3.555692] usb 3-1: New USB device found, idVendor=0424, idProduct=2744, bcdDevice= 2.21
[    3.563875] usb 3-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    3.571022] usb 3-1: Product: USB2744
[    3.574680] usb 3-1: Manufacturer: Microchip Tech
[    3.618592] hub 2-1:1.0: USB hub found
[    3.622463] hub 3-1:1.0: USB hub found
[    3.626229] hub 2-1:1.0: 3 ports detected
[    3.630257] hub 3-1:1.0: 3 ports detected
[    3.683235] usb 4-1: new SuperSpeed USB device number 2 using xhci-hcd
[    3.707436] usb 4-1: New USB device found, idVendor=0424, idProduct=5744, bcdDevice= 2.21
[    3.715611] usb 4-1: New USB device strings: Mfr=2, Product=3, SerialNumber=0
[    3.722740] usb 4-1: Product: USB5744
[    3.726398] usb 4-1: Manufacturer: Microchip Tech
[    3.795292] hub 4-1:1.0: USB hub found
[    3.799194] hub 4-1:1.0: 2 ports detected
[    3.803072] usb 1-1.1: new high-speed USB device number 3 using xhci-hcd
[    3.913377] usb 1-1.1: New USB device found, idVendor=0424, idProduct=2240, bcdDevice= 1.98
[    3.921737] usb 1-1.1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    3.929042] usb 1-1.1: Product: Ultra Fast Media 
[    3.933744] usb 1-1.1: Manufacturer: Generic
[    3.938012] usb 1-1.1: SerialNumber: 000000225001
[    3.943229] usb-storage 1-1.1:1.0: USB Mass Storage device detected
[    3.949803] scsi host0: usb-storage 1-1.1:1.0
[    3.971080] usb 3-1.3: new high-speed USB device number 3 using xhci-hcd
[    4.035071] usb 1-1.4: new high-speed USB device number 4 using xhci-hcd
[    4.075764] usb 3-1.3: New USB device found, idVendor=0424, idProduct=2740, bcdDevice= 2.00
[    4.084125] usb 3-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    4.091434] usb 3-1.3: Product: Hub Controller
[    4.095867] usb 3-1.3: Manufacturer: Microchip Tech
[    4.139748] usb 1-1.4: New USB device found, idVendor=0424, idProduct=2740, bcdDevice= 2.00
[    4.148098] usb 1-1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    4.155410] usb 1-1.4: Product: Hub Controller
[    4.159848] usb 1-1.4: Manufacturer: Microchip Tech
[    4.959835] scsi 0:0:0:0: Direct-Access     Generic  Ultra HS-COMBO   1.98 PQ: 0 ANSI: 0
[    4.969189] sd 0:0:0:0: [sda] 62410752 512-byte logical blocks: (32.0 GB/29.8 GiB)
[    4.977351] sd 0:0:0:0: [sda] Write Protect is off
[    4.982727] sd 0:0:0:0: [sda] No Caching mode page found
[    4.988036] sd 0:0:0:0: [sda] Assuming drive cache: write through
[    4.998107]  sda: sda1 sda2
[    5.002738] sd 0:0:0:0: [sda] Attached SCSI removable disk
[    5.024085] EXT4-fs (sda2): mounted filesystem with ordered data mode. Opts: (null). Quota mode: none.
[    5.033425] VFS: Mounted root (ext4 filesystem) on device 8:2.
```

Sorry, I do not have a KR260 and cannot immediately determine the cause.

I would like to know what kind of message I will get in the official ubuntu image.

If anyone looking at this issue can help me I would be glad.

**Comment by abyszuk** [comment]
Thank you for your fast response.
I'm attaching full kernel log outputs for both upstream Ubuntu and your version.
Please let me know if I can do more to help diagnose this.
[ours_dmesg.log](https://github.com/ikwzm/ZynqMP-FPGA-Ubuntu22.04-Desktop/files/13248104/ours_dmesg.log)
[upstream_dmesg.log](https://github.com/ikwzm/ZynqMP-FPGA-Ubuntu22.04-Desktop/files/13248105/upstream_dmesg.log)

**Comment by ikwzm** [comment]
Thank you very much.

If you have time, could you try ZynqMP-FPGA-Ubuntu22.04-Desktop v1.1.2?

https://github.com/ikwzm/ZynqMP-FPGA-Ubuntu22.04-Desktop/tree/v1.1.2

v1.2.0 uses Linux Kernel 6.1, while v1.1.2 uses Linux Kernel 5.15, the same as the official image.

**Comment by abyszuk** [comment]
Sure, I'll try it today or tomorrow.
Can I mix the kernel from v1.1.2 with rootfs v1.2.0?

**Comment by ikwzm** [comment]
The linux kernel module included in rootfs is different between v1.2.0 and v1.1.2 due to the different linux kernel versions.
Specifically, the installed linux image package is different.
v1.2.0 has linux-image-6.1.42-zynqmp-fpga-trial_6.1.42-zynqmp-fpga-trial-1_arm64.deb installed.
v1.1.2 has linux-image-5.15.108-zynqmp-fpga-trial_5.15.108-zynqmp-fpga-trial-2_arm64.deb installed.

If these linux image packages are not installed, device drivers built as kernel modules will not work.
But, conversely, if you do not use device drivers built as kernel modules, rootfs without the linux image packages installed may boot.

If you succeeded in booting v1.2.0 rootfs with v1.1.2 linux kernel, you can download the linux kernel image package from the following URL and install it later.

https://github.com/ikwzm/ZynqMP-FPGA-Ubuntu22.04-Desktop/tree/v1.1.2/debian

**Comment by ikwzm** [comment]
v3.0.0 has been released. 
This includes Linux Kernel 6.1.70 booted with KR260.
